lib/vrl/stdlib/src/parse_query_string.rs

```diff
@@ -0,0 +1,131 @@
+use bytes::Buf;
+use std::collections::btree_map::Entry;
+use std::collections::BTreeMap;
+use url::form_urlencoded;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseQueryString;
+
+impl Function for ParseQueryString {
+    fn identifier(&self) -> &'static str {
+        "parse_query_string"
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "parse query string",
+            source: r#"parse_query_string("foo=1&bar=2")"#,
+            result: Ok(r#"
+                {
+                    "foo": "1",
+                    "bar": "2"
+                }
+            "#),
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        Ok(Box::new(ParseQueryStringFn { value }))
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::BYTES,
+            required: true,
+        }]
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseQueryStringFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ParseQueryStringFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let query_string = self.value.resolve(ctx)?.try_bytes()?;
+        let mut result: BTreeMap<String, Value> = BTreeMap::new();
+        let parsed = form_urlencoded::parse(query_string.bytes());
```

Comment:
```suggestion
        let parsed = form_urlencoded::parse(&query_string);
```

Dereferencing `Byte` type results in `&[u8]`, which should work here.

Also, this function got [removed](https://docs.rs/bytes/1.0.1/bytes/struct.Bytes.html#method.bytes) in the 1.x release (we're still on 0.6 until Tokio is upgraded), so it might be good to just do that straight away.

Reply:
Thanks, corrected.